lightning/src/ln/chanmon_update_fail_tests.rs

```diff
@@ -2215,3 +2215,97 @@ fn channel_holding_cell_serialize() {
 	do_channel_holding_cell_serialize(true, false);
 	do_channel_holding_cell_serialize(false, true); // last arg doesn't matter
 }
+
+#[derive(PartialEq)]
+enum HTLCStatusAtDupClaim {
+	Received,
+	HoldingCell,
+	Cleared,
+}
+fn do_test_reconnect_dup_htlc_claims(htlc_status: HTLCStatusAtDupClaim) {
+	// When receiving an update_fulfill_htlc message, we immediately forward the claim backwards
+	// along the payment path before waiting for a full commitment_signed dance. This is great, but
+	// can cause duplicative claims if a node sends an update_fulfill_htlc message, disconnects,
+	// reconnects, and then has to re-send its update_fulfill_htlc message again.
+	// In previous code, we didn't handle the double-claim correctly, spuriously closing the
+	// channel on which the inbound HTLC was received.
```

Comment:
Are you saying that previous to this PR, if we did receive a duplicate `update_fulfiil_htlc` on the forward channel, we would have close the backward one ? You're referring to current `get_update_fulfill_htlc` or another behavior? Don't get it here...

Reply:
> Are you saying that previous to this PR, if we did receive a duplicate update_fulfiil_htlc on the forward channel, we would have close the backward one

Yes, that's correct. We'd `debug_assert!(false, ...` then close the channel when debug assertions were off.